Frameworks/UIKit/UIApplicationMain.mm

```diff
@@ -329,14 +333,31 @@ void _UIApplicationShutdown() {
     [[NSUserDefaults _standardUserDefaultsNoInitialize] _resumeSynchronize];
 }
 
-extern "C" void UIApplicationMainHandleToastNotificationEvent(const char* notificationData) {
-    NSString* data = Strings::IsEmpty<const char*>(notificationData) ? nil : [[NSString alloc] initWithCString:notificationData];
-    [[UIApplication sharedApplication] _sendNotificationReceivedEvent:data];
+extern "C" void UIApplicationMainHandleToastActionEvent(HSTRING toastArgument, IInspectable* toastUserInput) {
+    NSString* argument = Strings::WideToNSString(toastArgument);
+
+    // Convert IPropertyValue to NSString
+    WFCValueSet* values = [WFCValueSet createWith:toastUserInput];
+    NSMutableDictionary* userInput = [NSMutableDictionary new];
+    for (NSString* key in [values allKeys]) {
+        RTObject* holderObject = [values objectForKey:key];
+        ComPtr<IPropertyValue> value;
+        HRESULT result = holderObject.comObj.As(&value);
+        THROW_NS_IF_FAILED(result);
```

Comment:
Instead of caching the HRESULT in a temporary, you can `THROW_NS_IF_FAILED(holderObject.xxx)`